web/src/main/kotlin/no/kartverket/matrikkel/bygning/routes/v1/intern/egenregistrering/EgenregistreringRequest.kt

```diff
@@ -170,3 +193,231 @@ fun EgenregistreringRequest.toEgenregistrering(eier: String): Egenregistrering =
         prosess = ProsessKode.Egenregistrering,
         bruksenhetRegistrering = this.toBruksenhetRegistrering(),
     )
+
+@Serializable
+data class LeggTilFeltEgenregistreringRequest(
+    val bruksenhetId: Long,
+    val leggTilFelt: FeltRegistreringRequest,
+)
+
+@Serializable
+data class KorrigerFeltEgenregistreringRequest(
+    val bruksenhetId: Long,
+    val korrigerFelt: FeltRegistreringRequest,
+)
+
+@Serializable
+sealed class FeltRegistreringRequest {
+    @Serializable
+    @SerialName("byggeaar")
+    @Name("byggeaar")
+    data class ByggeaarFeltRegistreringRequest(
+        val byggeaar: Int,
+        val kildemateriale: KildematerialeKode,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("bruksareal")
+    @Name("bruksareal")
+    data class BruksarealFeltRegistreringRequest(
+        val totaltBruksareal: Double,
+        val etasjeRegistreringer: List<EtasjeBruksarealRegistreringRequest>?,
+        val kildemateriale: KildematerialeKode,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("vannforsyning")
+    @Name("vannforsyning")
+    data class VannforsyningFeltRegistreringRequest(
+        val vannforsyning: VannforsyningKode,
+        val kildemateriale: KildematerialeKode,
+        val gyldighetsaar: Int,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("avlop")
+    @Name("avlop")
+    data class AvlopFeltRegistreringRequest(
+        val avlop: AvlopKode,
+        val kildemateriale: KildematerialeKode,
+        val gyldighetsaar: Int,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("harIkkeEnergikilder")
+    @Name("harIkkeEnergikilder")
+    data class HarIkkeEnergikilderFeltRegistreringRequest(
+        val kildemateriale: KildematerialeKode,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("energikilder")
+    @Name("energikilder")
+    data class EnergikilderFeltRegistreringRequest(
+        val energikilder: List<EnergikildeFeltDataRequest>,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    data class EnergikildeFeltDataRequest(
+        val energikilde: EnergikildeKode,
+        val kildemateriale: KildematerialeKode,
+        val gyldighetsaar: Int,
+    )
+
+    @Serializable
+    @SerialName("harIkkeOppvarming")
+    @Name("harIkkeOppvarming")
+    data class HarIkkeOppvarmingFeltRegistreringRequest(
+        val kildemateriale: KildematerialeKode,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    @SerialName("oppvarming")
+    @Name("oppvarming")
+    data class OppvarmingFeltRegistreringRequest(
+        val oppvarming: List<OppvarmingFeltDataRequest>,
+    ) : FeltRegistreringRequest()
+
+    @Serializable
+    data class OppvarmingFeltDataRequest(
+        val oppvarming: OppvarmingKode,
+        val kildemateriale: KildematerialeKode,
+        val gyldighetsaar: Int,
+    )
```

Comment:
```suggestion
    @Serializable
    @SerialName("oppvarming")
    @Name("oppvarming")
    data class OppvarmingFeltRegistreringRequest(
        val oppvarming: List<OppvarmingFeltDataRequest>,
    ) : FeltRegistreringRequest() {
        @Serializable
        data class OppvarmingFeltDataRequest(
            val oppvarming: OppvarmingKode,
            val kildemateriale: KildematerialeKode,
            val gyldighetsaar: Int,
        )
    }
```

synes det er litt tydeligere att dette ikke er en faktisk FeltRegistreringRequest om man strukturerer det sånn